pkg/kubefedctl/join.go

```diff
@@ -841,32 +893,20 @@ func populateSecretInHostCluster(clusterClientset, hostClientset kubeclient.Inte
 	// Get the secret from the joining cluster.
 	var secret *corev1.Secret
 	err := wait.PollImmediate(1*time.Second, serviceAccountSecretTimeout, func() (bool, error) {
-		sa, err := clusterClientset.CoreV1().ServiceAccounts(joiningNamespace).Get(
+		joiningClusterSASecret, err := clusterClientset.CoreV1().Secrets(joiningNamespace).Get(
 			context.Background(), saName, metav1.GetOptions{},
```

Comment:
Isn't the Secret's name derived by calling `util.ClusterServiceAccountTokenSecretName`?

```suggestion
			context.Background(),  util.ClusterServiceAccountTokenSecretName(joiningClusterName, hostClusterName), metav1.GetOptions{},
```